src/shared/helpers/OSNotificationDatabaseSerializer.ts

```diff
@@ -0,0 +1,117 @@
+import { NotificationClickEventInternal } from "../models/NotificationEvent";
+import { IOSNotification, IOSNotificationActionButton } from "../models/OSNotification";
+import { OutcomesNotificationClicked, OutcomesNotificationReceived } from "../models/OutcomesNotificationEvents";
+/**
+ * Purpose: This file defines mapping so any changes to the public
+ * classes don't break indexDb serialization.
+ * 
+ * Naming Rules:
+ *    1. All classes that convert types to / from the db
+ *    must end in Serialize.
+ *    2. All interfaces used with classes ending in Serialize
+ *    must end in Schema.
+ * 
+ * NOTE: Any change made to interfaces that end in Schema require
+ *       bumping the indexDb.ts version and creating a migration.
+ */
+
+// Purpose: Used to fire the 'click' event and open a page it's URL.
+//    * Only used if no pages are open for the website
+export interface NotificationClickForOpenHandlingSchema {
+  readonly id: string; // indexDb's keyPath
+  readonly heading?: string;
+  readonly content: string;
+  readonly data?: object;
+  readonly url?: string;
+  readonly rr: boolean;
+  readonly icon?: string;
+  readonly image?: string;
+  readonly tag?: string;
+  readonly badge?: string;
+  readonly action?: string;
+  readonly buttons?: NotificationButtonsClickForOpenHandlingSchema[];
+  readonly timestamp: number;
+}
+
+interface NotificationButtonsClickForOpenHandlingSchema {
+  readonly action: string;
+  readonly title: string;
+  readonly icon?: string;
+  readonly url?: string;
+}
+
+export class NotificationClickForOpenHandlingSerializer {
```

Comment:
add comment here to check the top of the file for more details.
* This way you see hint in VS code / yoru IDE.